python/taichi/lang/matrix.py

```diff
@@ -1431,10 +1473,13 @@ class _MatrixFieldElement(_IntermediateMatrix):
         indices (taichi_python.ExprGroup): Indices of the element.
     """
     def __init__(self, field, indices):
-        super().__init__(field.n, field.m, [
-            expr.Expr(ti_python_core.subscript(e.ptr, indices))
-            for e in field._get_field_members()
-        ])
+        super().__init__(
+            field.n,
+            field.m, [
+                expr.Expr(ti_python_core.subscript(e.ptr, indices))
+                for e in field._get_field_members()
+            ],
+            ndim=getattr(field, "ndim", 2))
```

Comment:
Was thinking if we can default the value of `ndim` to 2 inside `field` construction, instead of checking `ndim`'s existence every-time before usage?

Reply:
iirc `ndim` is only used in Vector and Matrix fields, so when `field` is a StructField, then there's no need to declare an `ndim` for it.

Reply:
I see... I think the root reason is that essentially `ndim` is not part of the attributes for a `field` object.

Let's keep it this way for now, and try to resolve it in the refactored Matrix/Vec implementation.